backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala

```diff
@@ -497,12 +496,16 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
    * @return
    */
   override def genExtendedColumnarPostRules(): List[SparkSession => Rule[SparkPlan]] = {
-    val Array(major, minor, _) = SparkShimLoader.getSparkShims.getShimDescriptor.toString.split('.')
-    if (major.toInt > 3 || (major.toInt == 3 && (minor.toInt >= 4))) {
-      List()
-    } else {
-      List(spark => NativeWritePostRule(spark))
-    }
+
+    List(
+      spark =>
+        if (SparkShimLoader.getSparkShims.getExtendedColumnarPostRules(spark).nonEmpty) {
+          SparkShimLoader.getSparkShims.getExtendedColumnarPostRules(spark).get
+        } else {
+          new Rule[SparkPlan] {
+            override def apply(plan: SparkPlan): SparkPlan = plan
+          }
+        })
```

Comment:
Could it be better with something like

```suggestion
SparkShimLoader.getSparkShims.getExtendedColumnarPostRules(spark) ::: List() 
```

Then in future we could add more rules that are common to all Spark versions into the rhs list.